wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,129 @@
+# WG Node Lifecycle Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [wg-governance].
+
+[Kubernetes Charter README]: /committee-steering/governance/README.md
+
+## Scope
+
+The Kubernetes ecosystem currently faces challenges in node maintenance scenarios, with multiple
+projects independently addressing similar issues. The goal of this working group is to develop
+unified APIs that the entire ecosystem can depend on, reducing the maintenance burden across
+projects and addressing scenarios that impede node drain or cause improper pod termination. Our
+objective is to create easily configurable, out-of-the-box solutions that seamlessly integrate with
+existing APIs and behaviors. We will strive to make these solutions minimalistic and extensible to
+support advanced use cases across the ecosystem.
+
+To properly solve the node drain, we must first understand the node lifecycle. This includes
+provisioning/sunsetting of the nodes, PodDisruptionBudgets, API-initiated eviction and node
+shutdown. This then impacts both the node and pod autoscaling, load balancing, and the applications
+running in the cluster. All of these areas have issues and would benefit from a unified approach.
+
+### In scope
+
+- Explore a unified way of draining the nodes and managing node maintenance by introducing new APIs
```

Comment:
Is it worth including something about DRA device taints/drains?

Reply:
Seems relevant to me as this affects the pod and device/node lifecycle. @pohly what do you think about including and discussing https://github.com/kubernetes/enhancements/issues/5055 in the WG?

Reply:
Part of the motivation for 5055 is to *not* have to drain entire nodes when doing some maintenance which only affects one device or hardware managed by one driver - unless of course that maintenance ends up with having to reboot the node.

So I guess it depends?

Reply:
We would also like to handle the pod lifecycle better in any descheduling scenario (not just Node Drain/Maintenance). One option is to use the [EvictionRequest API](https://github.com/kubernetes/enhancements/issues/4563) which should give more power to the applications that are being disrupted.  So it might be interesting to see if we can make the disruption more graceful  in 5055.

Reply:
> Part of the motivation for 5055 is to not have to drain entire nodes when doing some maintenance which only affects one device or hardware managed by one driver - unless of course that maintenance ends up with having to reboot the node.

The other end of this to consider though is devices that span multiple nodes

Reply:
I have added it to the WG. I think it would be good to discuss this feature and its impact . Also, it might be better to hold off beta for some time.